iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalConstant.java

```diff
@@ -25,7 +25,7 @@ public class DockerLexicalConstant {
 
   public static final String COMMENT = "(?:" + LexicalConstant.SINGLE_LINE_COMMENT_HASH + ")";
   public static final String STRING_LITERAL_WITH_QUOTES = "\"(?:[^\"\\\\]*+(?:\\\\[\\s\\S])?+)*+\"";
-  public static final String STRING_LITERAL_WITHOUT_QUOTES = "[^\\s]+";
+  public static final String STRING_LITERAL_WITHOUT_QUOTES = "[^\\s]+(?<!\\\\)";
```

Comment:
In case we don't want to use the lookahead style for some reason, we could also setup this : `(?:\\\\*[^\\s\\\\]+)+`